persistence-jdbc/src/main/scala/zio/actors/persistence/jdbc/JDBCJournal.scala

```diff
@@ -0,0 +1,77 @@
+package zio.actors.persistence.jdbc
+
+import cats.effect.Blocker
+import com.zaxxer.hikari.HikariDataSource
+import doobie._
+import doobie.hikari.HikariTransactor
+import doobie.implicits._
+
+import zio.{ DefaultRuntime, IO, Promise, Task, ZIO }
+import zio.actors.ActorSystemUtils
+import zio.actors.persistence.PersistenceId.PersistenceId
+import zio.actors.persistence.journal.Journal
+import zio.actors.persistence.jdbc.JDBCConfig.DbConfig
+import zio.blocking.Blocking
+import zio.interop.catz._
+
+private[actors] object JDBCJournal {
+
+  private lazy val runtime           = new DefaultRuntime {}
+  private lazy val transactorPromise = runtime.unsafeRun(Promise.make[Exception, HikariTransactor[Task]])
+
+  def makeTransactor(dbConfig: DbConfig): ZIO[Blocking, Throwable, HikariTransactor[Task]] =
+    ZIO.runtime[Blocking].flatMap { implicit rt =>
+      for {
+        transactEC <- rt.environment.blocking.blockingExecutor.map(_.asEC)
+        connectEC  = rt.platform.executor.asEC
+        ds         = new HikariDataSource()
+        _          <- IO.effect(Class.forName("org.postgresql.Driver"))
+        _          = ds.setJdbcUrl(dbConfig.dbURL.value)
+        _          = ds.setUsername(dbConfig.dbUser.value)
+        _          = ds.setPassword(dbConfig.dbPass.value)
+        transactor <- IO.effect(HikariTransactor.apply[Task](ds, connectEC, Blocker.liftExecutionContext(transactEC)))
+      } yield transactor
+    }
+
+  def transactor(dbConfig: DbConfig): Task[HikariTransactor[Task]] =
+    transactorPromise.poll.flatMap {
+      case Some(value) => value
+      case None =>
+        for {
+          newTnx <- makeTransactor(dbConfig).provide(Blocking.Live)
+          _      <- transactorPromise.succeed(newTnx)
+        } yield newTnx
+    }
+
+  def getJournal[Ev](actorSystemName: String, configStr: String): Task[JDBCJournal[Ev]] =
+    for {
+      dbConfig <- JDBCConfig.getDbConfig(actorSystemName, configStr)
+      tnx      <- transactor(dbConfig)
+    } yield new JDBCJournal[Ev](tnx)
+
+}
+
+private[actors] class JDBCJournal[Ev](tnx: Transactor[Task]) extends Journal[Ev] {
```

Comment:
Please make it final. Also, the constructor should probably be private. One other thing, please prefer listing the class *before* its companion.